letsencrypt/tests/proof_of_possession_test.py

```diff
@@ -78,6 +80,15 @@ def test_perform_with_input(self, mock_input):
         self.assertFalse(self.proof_of_pos.perform(self.achall))
         self.assertTrue(self.proof_of_pos.perform(self.achall).verify())
 
+    def test_perform_with_key_dir(self):
+        # Remove the matching certificate
+        self.installer.get_all_certs_keys.return_value.pop()
+
+        self.config.key_dir = tempfile.mkdtemp('key_dir')
+        crypto_util.init_save_key(2048, self.config.key_dir)
```

Comment:
There's no need to generate a new key here. `CERT2_KEY_PATH` and `CERT3_KEY_PATH` are paths to already generated private keys. `CERT3_KEY_PATH` is the one that matches the public key for the challenge used in these tests.